src/challenger-agent.ts

```diff
@@ -0,0 +1,111 @@
+import {ChallengeManager, expectedNumOfLeaves, State, stepForIndex} from './challenge-manager';
+import {fingerprint, Role} from './tests/challenge-manager.test';
+import {generateWitness, WitnessProof} from './merkle';
+import {DisputeGame} from './dispute-game';
+
+/**
+ * The ChallengerAgent is a dispute game participant. The agent is initialized with a set of states.
+ * The agent uses the states to take turns in the dispute game.
+ */
+export class ChallengerAgent {
+  private cm: ChallengeManager;
+  constructor(
+    private role: Role,
+    private states: State[],
+    numSplits: number,
+    disputeGame: DisputeGame
```

Comment:
Shouldn't the `ChallengerAgent` not know about the `DisputeGame`  since the `DisputeGame` is just used for testing?

Reply:
That's a good point. At the moment, `DisputeGame` holds the global context for the dispute. In the prototype, that context includes the `ChallengeManager`. In production, global context will be stored on-chain.

Happy to change the constructor api if you have ideas!

Reply:
Couldn't we pass in just the `ChallengeManager` as that should contain the global context? That seems like it would be closer to what our eventual API will look like.

Reply:
`ChallengeManager` constructor currently accepts the initial set of states for the dispute game. So it seems appropriate for the `ChallangerAgent` to "deploy" the contract since the challenger agent is the entity that tracks states for a player.

The challenger agent needs access to some global context to assign the deployed challenger manager to that context. Curious if that design makes sense to you!

Reply:
I think that makes sense! However I wonder if we could have the `ChallengerAgent` accept a **optional** ChannelManager instance (instead of a `DisputeGame`)? If there is no channel manager it constructs one (mimicking deploying the contract) and if there is one it uses that. This would prevent the agent having to know about the `DisputeGame` type at all.

```
const challengeAgent = new ChallengerAgent({channelManager:undefined});
const responderAgent = new ChallengerAgent({channelManager:challengeAgent.cm});

```

Reply:
That's definitely an option! I am open to it, though I prefer the global context approach. We could also separate the global context from the `DisputeGame` object if that is cleaner.

Reply:
> That's definitely an option! I am open to it, though I prefer the global context approach. We could also separate the global context from the `DisputeGame` object if that is cleaner.

I think I'd be prefer if the global context was separated from the `DisputeGame`. Right now the `DisputeGame` seems to be both a test helper and a global context store.